src/pages/index.tsx

```diff
@@ -129,12 +129,10 @@ export default function HomePage({ schedule }: Props) {
       );
     }
 
-    return schedule.filter(s => {
-      return s[searchBy].toLowerCase().includes(searchKeyword.toLowerCase());
-    });
+    return filteredSchedules;
     // Resorting to a typecast here for a quick workaround.
     // TODO: Type this function better
-  }, [schedule, searchBy, searchKeyword, userLocation]) as VaccinationDataWithDistance[];
+  }, [schedule, searchKeyword, filtered, userLocation]) as VaccinationDataWithDistance[];
```

Comment:
just realized that this memo is not ideal, too many computed values (`filtered` is computed value from `schedule` and `searchKeyword`). let's fix this in the separate PR

Reply:
yeah, i'm not that familiar with React and Typescript, so maybe you or the others can tackle optimizing this later.